### PR TITLE
use CPM.cmake module to fetch project_options

### PR DESCRIPTION
**Pull request by ClausKlein**
ENABLE_CONAN by default to test conan with CPM fetch contents
disable gui dependency for simlicity (and reduce buildtime)

see too https://github.com/aminya/project_options/pull/116

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -22,11 +22,20 @@ set(OPT_WARNINGS_AS_ERRORS_DEVELOPER_DEFAULT TRUE)
...
+# FetchContent_Declare(_project_options
+#                      URL https://github.com/cpp-best-practices/project_options/archive/refs/tags/v0.17.0.zip)
+# FetchContent_MakeAvailable(_project_options)
+# include(${_project_options_SOURCE_DIR}/Index.cmake)
+
+# ---- Add dependencies via CPM ----
+# see https://github.com/cpm-cmake/CPM.cmake for more info
+include(cmake/CPM.cmake)
+
+# Add project_options v0.18.0, but my version! CK
```
Comment (ClausKlein) [inline CMakeLists.txt:35]: only for test now, later original may be used!

File: CMakeLists.txt [hunk 2]
```diff
@@ -92,6 +101,7 @@ dynamic_project_options(
   PCH_HEADERS
   <vector>
   <string> # This is a list of headers to pre-compile, here are some common ones
+  ENABLE_CONAN
```
[the remark this answers is not in the document]
Comment (ddalcino) [reply CMakeLists.txt:104]: This is probably an artifact of the split between cpp_boilerplate_project and cpp_starter_project. Without `ENABLE_CONAN` here, this project cannot build at all.

File: test/CMakeLists.txt [hunk 3]
```diff
@@ -1,6 +1,6 @@
 find_package(Catch2 REQUIRED)
 
-include(CTest)
+#XXX include(CTest)
```
[the remark this answers is not in the document]
Comment (ddalcino) [reply test/CMakeLists.txt:3]: https://cmake.org/cmake/help/latest/module/CTest.html

[the remark this answers is not in the document]
Comment (ClausKlein) [reply test/CMakeLists.txt:3]: you want to enable submissions to a CDash server?

**Comment by ClausKlein** [comment]
This PR was only to tigger the ci test.
It shows the possibility to use CPM cmake module to fetch the `project_options`

**Comment by ddalcino** [comment]
> This PR was only to tigger the ci test.

There are other ways to trigger a ci test. See GH actions syntax here: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#onevent_nametypes

I often add `workflow_dispatch` to my `.github/workflows/ci.yml` file, so I can manually trigger a workflow via a button in the Github UI. A lot of projects have their workflows set up to run workflows on push to any branch, not just main.

**Pull request closed without merging**